src/riot-rs-macros/src/thread.rs

```diff
@@ -58,17 +58,13 @@ pub fn thread(args: TokenStream, item: TokenStream) -> TokenStream {
 
     let expanded = quote! {
         #no_mangle_attr
-        #[inline(always)]
         #thread_function
 
         #[#riot_rs_crate::linkme::distributed_slice(#riot_rs_crate::thread::THREAD_FNS)]
         #[linkme(crate = #riot_rs_crate::linkme)]
         fn #slice_fn_name_ident() {
-            fn trampoline(_arg: ()) {
-                #fn_name();
-            }
             let stack = #riot_rs_crate::static_cell::make_static!([0u8; #stack_size as usize]);
-            #riot_rs_crate::thread::thread_create(trampoline, (), stack, #priority);
+            unsafe { #riot_rs_crate::thread::thread_create_raw(#fn_name as usize, 0, stack, #priority)};
```

Comment:
it would, somewhat. but it would also pin the types. as is, *any* function pointer (no matter the expected arguments) can be used (by casting to usize).

Reply:
well, it enables us to use `thread_create_raw()` here with `fn()`, and in `thread_create()` with `fn(Arguable)`.